src/aquarium-optimized/Context.h

```diff
@@ -47,15 +53,40 @@ class Context
 
     int getClientWidth() const { return mClientWidth; }
     int getclientHeight() const { return mClientHeight; }
+    std::bitset<static_cast<size_t>(TOGGLE::TOGGLEMAX)> getDefaultToggleBitset()
+    {
+        return mDefaultToggleBitset;
+    }
 
     virtual Model *createModel(Aquarium *aquarium, MODELGROUP type, MODELNAME name, bool blend) = 0;
 
     virtual void initGeneralResources(Aquarium* aquarium) {}
     virtual void updateWorldlUniforms(Aquarium* aquarium) {}
 
+    std::string getPropPlacementPath() const { return resourceHelper->getPropPlacementPath(); }
```

Comment:
Use `const std::string &getPropPlacementPath() const` and same with below.

Reply:
Is it better if we expose getResourceHelper? Then in other class, we can directly use like below:
```
const ResourceHelper* helper = context->getResourceHelper();
helper->getPropPlacementPath()
...